framework/wazuh/core/cluster/common.py

```diff
@@ -484,6 +489,141 @@ async def send_request(self, command: bytes, data: bytes) -> bytes:
             return b'Error sending request: timeout expired.'
         return response_data
 
+    async def get_chunks_in_task_id(self, task_id: bytes, error_command: bytes) -> dict:
+        """Function in charge of collecting the chunks stored under task_id.
+
+        Parameters
+        ----------
+        task_id : bytes
+            Pre-defined task_id to identify this object.
+        error_command : bytes
+            Command sent to the sender node in case of error.
+
+        Returns
+        -------
+        data : dict
+            Chunks collected through task_id.
+        """
+        try:
+            # Chunks were stored under 'task_id' as an string.
+            received_string = self.in_str[task_id].payload
+            data = json.loads(received_string.decode())
+        except KeyError as e:
+            print(str(e))
+            await self.send_request(command=error_command,
+                                    data=f'error while trying to access string under task_id {str(e)}.'.encode())
+            raise exception.WazuhClusterError(3035,
+                                              extra_message=f"it should be under task_id {str(e)}, but it's empty.")
+        except ValueError as e:
+            await self.send_request(command=error_command, data=f'error while trying to load JSON: {str(e)}'.encode())
+            raise exception.WazuhClusterError(3036, extra_message=str(e))
+
+        return data
+
+    async def update_chunks_wdb(self, data: dict, info_type: str, logger: logging.Logger, error_command: bytes) -> dict:
+        """Send the received data to WDB and returns the result of the operation.
+
+        Parameters
+        ----------
+        data : dict
+            Dict containing command and list of chunks to be sent to wazuh-db.
+        info_type : str
+            Information type handled.
+        logger : Logger object
+            Logger to use.
+        error_command : bytes
+            Command sent to the sender node in case of error.
+
+        Returns
+        -------
+        result : dict
+            Dict containing number of updated chunks, error messages (if any) and time spent.
+        """
+        try:
+            result = await cluster.run_in_pool(self.loop, self.server.task_pool, send_data_to_wdb, data,
+                                               self.cluster_items['intervals']['worker']['timeout_agent_groups'],
```

Comment:
Yes, this is to avoid having to change a good part of the references. In any case, in the near future we should refactor this part.

Reply:
Right now it is separated in two, one for master and one for worker to make this more configurable. It is only used in common.py but the common.py object is started from the worker.py or from the master.py so it could have different values.

Reply:
Uhm I see. However, the `cluster.json` is the same in all nodes so the value of `self.cluster_items['intervals']['worker']['timeout_agent_groups']` will be the same regardless of whether it is used in the master or in a worker.

In fact, I think `self.cluster_items['intervals']['master']['timeout_agent_groups']` isn't even being used, only the value defined for the workers is. 

Maybe there is something I don't understand or I'm not taking into account, let me know in that case.